src/main/java/dev/codesoapbox/dummy4j/dummies/shared/string/Insertion.java

```diff
@@ -0,0 +1,44 @@
+package dev.codesoapbox.dummy4j.dummies.shared.string;
+
+import java.util.StringJoiner;
+import java.util.regex.Pattern;
+
+/**
+ * Provides convenience methods for inserting characters into strings
+ *
+ * @since SNAPSHOT
+ */
+public final class Insertion {
+
+    /**
+     * The {@code (?<=\G.{4})} pattern explanation:
+     * <p>
+     * <ul>
+     *   <li>
+     *     {@code (?<=...)} - positive lookbehind, ensures that the given pattern will match, ending at the current
+     *     position in the expression. The pattern must have a fixed width. Does not consume any characters.
+     *   </li>
+     *   <li>
+     *     {@code \G} - takes position at the end of the previous match or the start of the string for the first match.
+     *   </li>
+     *   <li>
+     *     {@code .{4}} - matches any character exactly 4 times.
+     *   </li>
+     * </ul>
+     */
+    private static final Pattern SPLIT_EVERY_FOUR_CHARS_PATTERN = Pattern.compile("(?<=\\G.{4})");
+
+    private Insertion() {
+    }
+
+    public static String insertEveryFourCharacter(String input, String insertion) {
```

Comment:
`fourth`

+ I think we could make this more efficient by manually adding the separator, instead of relying on regex:
```java
        int everyN = 4;
        char[] input = "testtesttest".toCharArray();
        char separator = '-';
        StringBuilder stringBuilder = new StringBuilder();

        for (int i = 0; i < input.length; i++) {
            stringBuilder.append(input[i]);
            if((i+1) % everyN == 0 && i < input.length - 1) {
                stringBuilder.append(separator);
            }
        }

        System.out.println(stringBuilder.toString());
```

This also allows us to make the method `insertEveryNthCharacter`.